app/helpers/pageflow/social_share_helper.rb

```diff
@@ -32,14 +32,16 @@ def social_share_page_description(page)
     def social_share_entry_image_tags(entry)
       image_urls = []
 
-      image_urls << ImageFile.find(entry.share_image_id).thumbnail_url(:medium) if entry.share_image_id.present?
-
-      entry.pages.each do |page|
-        if image_urls.size >= 4
-          break
-        else
-          image_urls << page.thumbnail_url(:medium)
-          image_urls.uniq!
+      if entry.share_image_id.present?
+        image_urls << ImageFile.find(entry.share_image_id).thumbnail_url(:medium)
+      else
+        entry.pages.each do |page|
+          if image_urls.size >= 4
+            break
+          else
+            image_urls << page.thumbnail_url(:medium)
+            image_urls.uniq!
+          end
         end
       end
```

Comment:
Can we add a test here? I'm surprised we didn't so far, given the complexity of the logic.